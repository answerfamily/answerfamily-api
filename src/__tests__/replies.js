const {
  loadESFixtures,
  unloadESFixtures,
  loadMongoFixtures,
  unloadMongoFixtures,
} = require('../../test/fixtures');
const gql = require('../../test/gql');
const {
  ES_FIXTURES,
  MONGO_FIXTURES,
} = require('../__fixtures__/articleParagraphReply');

describe('replies', () => {
  beforeAll(async () => {
    await loadESFixtures(ES_FIXTURES);
    await loadMongoFixtures(MONGO_FIXTURES);
  });

  afterAll(async () => {
    await unloadESFixtures(ES_FIXTURES);
    await unloadMongoFixtures(MONGO_FIXTURES);
  });

  it('lists unfiltered replies', async () => {
    const { data, errors } = await gql`
      {
        replies {
          id
          text
        }
      }
    `();

    expect(errors).toBeUndefined();
    expect(data).toMatchSnapshot();
  });

  it('does limiting', async () => {
    const { data, errors } = await gql`
      {
        replies(first: 1) {
          id
          text
        }
      }
    `();

    expect(errors).toBeUndefined();
    expect(data).toMatchSnapshot();
  });

  it('does skipping', async () => {
    const { data, errors } = await gql`
      {
        replies(skip: 1) {
          id
          text
        }
      }
    `();

    expect(errors).toBeUndefined();
    expect(data).toMatchSnapshot();
  });

  it('does sorting', async () => {
    const { data, errors } = await gql`
      {
        replies(sort: [{ by: createdAt, order: ASC }]) {
          id
          text
        }
      }
    `();

    expect(errors).toBeUndefined();
    expect(data).toMatchSnapshot();
  });

  it('does filtering', async () => {
    const { data, errors } = await gql`
      {
        replies(filter: { contain: "保守團體的打壓 國小的課本" }) {
          id
          text
        }
      }
    `();

    expect(errors).toBeUndefined();
    expect(data).toMatchSnapshot();
  });
});
