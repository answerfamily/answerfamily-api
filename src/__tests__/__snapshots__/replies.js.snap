// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`replies does filtering 1`] = `
Object {
  "replies": Array [
    Object {
      "id": "r2",
      "text": "國中小的「同志教育」其實一直都在《性別平等教育法實行細則》裡頭，只是在保守團體的打壓之下，在國小的課本中沒有出現過。",
    },
  ],
}
`;

exports[`replies does limiting 1`] = `
Object {
  "replies": Array [
    Object {
      "id": "r1",
      "text": "「外國同性戀藉結婚來台治療愛滋病，都是健保支付，全民買單」這句話其實並不算錯，但實務上並不會有「拖垮健保」的狀況，也跟同性婚姻關係不大，所以不用太擔心唷。",
    },
  ],
}
`;

exports[`replies does skipping 1`] = `
Object {
  "replies": Array [
    Object {
      "id": "r2",
      "text": "國中小的「同志教育」其實一直都在《性別平等教育法實行細則》裡頭，只是在保守團體的打壓之下，在國小的課本中沒有出現過。",
    },
  ],
}
`;

exports[`replies does sorting 1`] = `
Object {
  "replies": Array [
    Object {
      "id": "r1",
      "text": "「外國同性戀藉結婚來台治療愛滋病，都是健保支付，全民買單」這句話其實並不算錯，但實務上並不會有「拖垮健保」的狀況，也跟同性婚姻關係不大，所以不用太擔心唷。",
    },
    Object {
      "id": "r2",
      "text": "國中小的「同志教育」其實一直都在《性別平等教育法實行細則》裡頭，只是在保守團體的打壓之下，在國小的課本中沒有出現過。",
    },
  ],
}
`;

exports[`replies lists unfiltered replies 1`] = `
Object {
  "replies": Array [
    Object {
      "id": "r1",
      "text": "「外國同性戀藉結婚來台治療愛滋病，都是健保支付，全民買單」這句話其實並不算錯，但實務上並不會有「拖垮健保」的狀況，也跟同性婚姻關係不大，所以不用太擔心唷。",
    },
    Object {
      "id": "r2",
      "text": "國中小的「同志教育」其實一直都在《性別平等教育法實行細則》裡頭，只是在保守團體的打壓之下，在國小的課本中沒有出現過。",
    },
  ],
}
`;
